core-default/src/main/java/me/aurium/beetle/defaults/task/CommonTaskRunner.java

```diff
@@ -0,0 +1,60 @@
+package me.aurium.beetle.defaults.task;
+
+import me.aurium.beetle.api.task.TaskRunner;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.function.Supplier;
+
+//taskrunner for platforms with a main thread to use
+public class CommonTaskRunner implements TaskRunner {
+
+    private final Logger logger = LoggerFactory.getLogger(CommonTaskRunner.class);
```

Comment:
```suggestion
    private static final Logger logger = LoggerFactory.getLogger(CommonTaskRunner.class);
```
Optional suggestion to make the logger static